subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/TaskIdentity.java

```diff
@@ -47,13 +47,8 @@
         this.uniqueId = uniqueId;
     }
 
-    /**
-     * Creates a task identity with a specific unique id.
-     * <p>
-     * This method should not be called directly, use {@link TaskIdentityFactory} instead.
-     */
-    public static <T extends Task> TaskIdentity<T> create(String name, Class<T> type, ProjectInternal project, long uniqueId) {
-        return new TaskIdentity<>(
+    TaskIdentity(String name, Class<T> type, ProjectInternal project, long uniqueId) {
+        this(
```

Comment:
🤔 Would it make sense to get rid of this constructor as well?

Reply:
I guess we might as well